openedx/features/genplus_features/genplus_learning/api/v1/serializers.py

```diff
@@ -203,9 +203,14 @@ class Meta:
     def get_username(self, obj):
         user = obj.gen_user.user
         profile = UserProfile.objects.filter(user=user).first() if user else None
-        if profile:
-            return profile.name
-        return obj.gen_user.email
+        name = ''
+        if profile and (profile.name or '').strip():
+            name = profile.name
+        elif (user.first_name or '').strip() or (user.last_name or '').strip():
+            name = f'{user.first_name} {user.last_name}'.strip()
+        else:
+            name = obj.gen_user.email
+        return name
 
     def get_has_first_logon(self, obj):
         if obj.gen_user.from_private_school:
```

Comment:
![image](https://github.com/GenPlusLeaders/edx-platform/assets/52816086/d73f5ef1-3651-43a4-964d-fd297a0426e4)



@qasimgulzar Here are all the places where we are using the **profile.name** for getting the full name.
I believe what we can do is move this name to the GenUser model level. maybe a property and then utilize that in the serializers or wherever it required.